llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -2459,6 +2459,20 @@ void VPlanTransforms::canonicalizeEVLLoops(VPlan &Plan) {
 
   VPBasicBlock *HeaderVPBB = EVLPhi->getParent();
   VPValue *EVLIncrement = EVLPhi->getBackedgeValue();
+  VPValue *AVL;
+  if (!match(EVLIncrement,
+             m_c_Add(m_ZExtOrSelf(m_EVL(m_VPValue(AVL))), m_Specific(EVLPhi))))
+    llvm_unreachable("Didn't find AVL?");
```

Comment:
hmm this is a bit unusual in the vectorizer code base, would it be much more code if this would be a match & assert + maybe_unusued?

Same below